### PR TITLE
add(scan): Add the `scan_start_where_left` test to CI

### LINKED ISSUE #8085: Add `scan_start_where_left` test to CI
In https://github.com/ZcashFoundation/zebra/pull/8080 we are adding a test called `scan_start_where_left`. This test uses a synchronized state so it is marked as `ignored`. We want to run this test in the CI in a similar way we do with other ignored tests.

This is blocked until  https://github.com/ZcashFoundation/zebra/pull/8080 is merged.

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation

Close #8085.

#### Check before marking the PR as ready for review:
  - [x] Will the PR name make sense to users?
  - [x] Does the PR have a priority label?
  - [x] Have you added or updated tests?
  - [x] Is the documentation up to date?

## Solution

- Add a new job named `scan-start-where-left-test` to the _Integration Tests on GCP_ workflow.

### Reviewer Checklist

Check before approving the PR:
  - [ ] Does the PR scope match the ticket?
  - [ ] Are there enough tests to make sure it works? Do the tests cover the PR motivation?
  - [ ] Are all the PR blockers dealt with?
        PR blockers can be dealt with in new tickets or PRs.

_And check the PR Author checklist is complete._

### CODE REVIEW

**Comment by upbqdn** [comment]
Let's see if the new job actually runs in the CI.

File: .github/workflows/ci-integration-tests-gcp.yml [hunk 1]
```diff
@@ -589,20 +589,46 @@ jobs:
...
+  scan-start-where-left-test:
+    name: Scan starts where left
+    needs: [test-full-sync, get-available-disks]
+    uses: ./.github/workflows/sub-deploy-integration-tests-gcp.yml
+    if: ${{ !cancelled() && !failure() && (fromJSON(needs.get-available-disks.outputs.zebra_tip_disk) || needs.test-full-sync.result == 'success') && github.event.inputs.regenerate-disks != 'true' && github.event.inputs.run-full-sync != 'true' && github.event.inputs.run-lwd-sync != 'true' }}
+    with:
+      app_name: zebrad
+      test_id: scan-start-where-left
+      test_description: Test that the scanner can continue scanning where it was left when zebrad restarts.
+      test_variables: "-e NETWORK=${{ inputs.network || vars.ZCASH_NETWORK }} -e TEST_SCAN_START_WHERE_LEFT=1 -e ZEBRA_FORCE_USE_COLOR=1 -e ZEBRA_CACHED_STATE_DIR=/var/cache/zebrad-cache -e TEST_FEATURES=shielded-scan"
```
Comment (arya2) [inline .github/workflows/ci-integration-tests-gcp.yml]:
```suggestion
      test_variables: "-e NETWORK=${{ inputs.network || vars.ZCASH_NETWORK }} -e TEST_SCAN_START_WHERE_LEFT=1 -e ZEBRA_FORCE_USE_COLOR=1 -e ZEBRA_CACHED_STATE_DIR=/var/cache/zebrad-cache"
```

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -326,6 +327,11 @@ case "$1" in
         check_directory_files "${ZEBRA_CACHED_STATE_DIR}"
         run_cargo_test "${ENTRYPOINT_FEATURES}" "submit_block"
 
+      elif [[ "${TEST_SCAN_START_WHERE_LEFT}" -eq "1" ]]; then
+        # Test that the scanner can continue scanning where it was left when zebrad restarts.
+        check_directory_files "${ZEBRA_CACHED_STATE_DIR}"
+        run_cargo_test "${ENTRYPOINT_FEATURES}" "scan_start_where_left"
```
Comment (arya2) [inline docker/entrypoint.sh]:
```suggestion
        run_cargo_test "shielded-scan" "scan_start_where_left"
```

**Review by arya2 — DISMISSED** [review]

**Review by arya2 — APPROVED** [review]

**Comment by upbqdn** [comment]
@arya2 thanks for the fix!

I looked at our Dockerfile, and I noticed we build the test binaries when building the image: https://github.com/zcashfoundation/zebra/blob/88dd18adb7db6e6127da578d9d7aaa7cab869fe9/docker/Dockerfile#L133-L134.

The new job in this PR runs the tests here https://github.com/zcashfoundation/zebra/blob/88dd18adb7db6e6127da578d9d7aaa7cab869fe9/docker/entrypoint.sh#L183 with a different set of features. That means the tests are rebuilt each time the job runs anyway. I wonder if we could omit building the tests in the Dockerfile.

Edit: The answer is no because only a small subset of the crates is recompiled. If we omitted the building in the Dockerfile, each job would need to rebuild all of the crates.

**Pull request merged**